tgui/packages/tgui/interfaces/GasCanister/Detonator.js

```diff
@@ -0,0 +1,210 @@
+      wireStatus,
+      time,
+      isPrimed,
+    },
+    onWireInteract,
+    onSetTimer,
+  } = props;
+
+  return (
+    <Flex>
+      <Flex.Item>
+        <LabeledList>
+          { wireNames.map((entry, i) => (
+            <LabeledList.Item
+              key={entry + i}
+              label={entry}>
+              <Box
+                height={1.7}>
+                { (wireStatus && wireStatus[i]) ? (
+                  <Fragment>
+                    <Button
+                      icon="cut"
+                      content="Cut"
+                      onClick={() => onWireInteract("cut", i)} />
+                    <Button
+                      icon="bolt"
+                      content="Pulse"
+                      onClick={() => onWireInteract("pulse", i)} />
+                  </Fragment>)
+                  : (
+                    <Box
+                      color="average"
+                      minHeight={1.4}>
+                      Cut
+                    </Box>) }
+              </Box>
+            </LabeledList.Item>
+          ))}
+        </LabeledList>
+      </Flex.Item>
+      <Flex.Item
+        mr={5}
+        mt={2} >
+        <Flex
+          direction="column"
+          align="center">
+          <Flex.Item>
+            <DetonatorTimer
+              time={time}
+              isPrimed={isPrimed} />
+          </Flex.Item>
+          <Flex.Item>
+            <Button
+              mt={1}
+              disabled={isPrimed}
+              icon="fast-backward"
+              onClick={() => onSetTimer(time - 300)} />
+            <Button
+              mt={1}
+              disabled={isPrimed}
+              icon="backward"
+              onClick={() => onSetTimer(time - 10)} />
+            <Button
+              mt={1}
+              disabled={isPrimed}
+              icon="forward"
+              onClick={() => onSetTimer(time + 10)} />
+            <Button
+              mt={1}
+              disabled={isPrimed}
+              icon="fast-forward"
+              onClick={() => onSetTimer(time + 300)} />
+          </Flex.Item>
+        </Flex>
+      </Flex.Item>
+    </Flex>
+  );
+};
+
+const DetonatorUtility = props => {
+  const {
+    detonator: {
+      isAnchored,
+      trigger,
+      safetyIsOn,
+      isPrimed,
+    },
+    detonatorAttachments,
+    onToggleAnchor,
+    onToggleSafety,
+    onPrimeDetonator,
+    onTriggerActivate,
+  } = props;
+
+  const renderArmingStatus = () => {
+    if (safetyIsOn) {
+      return ("The safety is on, therefore, you cannot prime the bomb.");
+    } else if (!safetyIsOn && !isPrimed) {
+      return (
+        <Button
+          color="danger"
+          icon="bomb"
+          content="Prime"
+          onClick={onPrimeDetonator} />);
+    } else {
+      return (
+        <Box
+          bold
+          color="red">
+          PRIMED
+        </Box>);
+    }
+  };
+
+  return (
+    <LabeledList>
+      <LabeledList.Item
+        className="utility-list-item"
```

Comment:
Prefer overly descriptive `className`, following [BEM](https://css-tricks.com/bem-101/) formatting (so `detonator-utility__list-item` in this case).

Reply:
done